Update codebase for annex v7

kyleam opened the pull request:
- [X] Add `supports_unlocked_pointers` method
- [X] Drop `known_failure_v6` decorator
- [X] Convert `skip_v6` to `skip_v6_or_later`
- [X] Other miscellaneous test adjustments for v7 repositories
- [X] Adjust comments and docstrings for v7
- [x] remove root commit (for testing latest annex)

Closes #2969.

kyleam commented:
> CLN+TST: Drop known_failure_v6 decorator 

Hmm, based on the test failures, looks like the crawler still depends on this.  I'll have a look at whether those marked tests are really still known failures.  If they are, I'll adjust  known_failure_v6 rather than dropping it.

yarikoptic commented:
I think it would be ok for a release cycle to leave alias for known_failure_v6 to deprecate it in the next one

I don't think we have bandwidth to support also previous versions of annex, so we should just boost minimal version for annex. I shouldn't forget to check/state in Debian package that new annex breaks current release of datalad

yarikoptic commented:
And with above said we should drop support for direct mode. But also see if we need to do anything to react appropriately whenever direct mode repo would get automatically upgraded by annex to v7. We might need a tarsal with direct mode repo to test

yarikoptic commented on datalad/support/annexrepo.py:
```diff
@@ -1152,6 +1152,16 @@ def is_crippled_fs(self):
         self.config.reload()
         return self.config.getbool("annex", "crippledfilesystem", False)
 
+    def supports_unlocked_pointers(self):
```
should may be it become a `@property`?

kyleam replied:
Fine by me.  I went with a plain method because `is_crippled_fs` right above it was.  Updated in push.

yarikoptic commented on datalad/support/tests/test_annexrepo.py:
```diff
@@ -1286,13 +1284,14 @@ def test_repo_version(path1, path2, path3):
         annex = AnnexRepo(path2, create=True)
         version = annex.repo.config_reader().get_value('annex', 'version')
-        eq_(version, 6)
+        assert_in(version, [6, 7])
```
would it may be make sense to hardcode our knowledge on annex behavior, i.e. that v7 would upgrade to 7? rationale: if we assume that now, and then for some reason annex changes behavior to retain v6, we might start to fail elsewhere without this test failing

kyleam replied:
I've added a TODO comment to update the check once we bump the git-annex dependency.

yarikoptic commented on datalad/support/tests/test_annexrepo.py:
```diff
@@ -1837,7 +1835,7 @@ def _test_status(ar):
     def sync_wrapper(push=False, pull=False, commit=False):
         # wraps common annex-sync call, since it currently fails under
-        # mysterious circumstances in V6 adjusted branch setups
+        # mysterious circumstances in V6+ adjusted branch setups
```
just for consistency
```suggestion
        # mysterious circumstances in v6+ adjusted branch setups
```

kyleam replied:
v6 versus V6 is already used inconsistently across the codebase.  Switching one instance doesn't improve the situation.

yarikoptic commented:
I verified that appveyor failures we are seeing seems to relate to annex switching to v7 (https://github.com/datalad/datalad/issues/2976)

yarikoptic commented:
having said [above](https://github.com/datalad/datalad/pull/2975#issuecomment-435715315) about not supporting previous annex releases, I think we should at least try to fix up to stay compatible with v7 for the sake of 0.11.1, and then go for "no direct" etc in 0.12.  @kyleam - did you spot any points which would somehow show that it wouldn't be possible?

kyleam commented:
>  @kyleam - did you spot any points which would somehow show that it wouldn't be possible?

Aside from the appveyor/direct mode failure being discussed elsewhere, I suppose?  I think we should be fine, but we should know once we drop the root commit of the branch, right?  That'll bring us back to the old annex.

kyleam commented:
> I think it would be ok for a release cycle to leave alias for known_failure_v6 to deprecate it in the next one

I've reverted the commit that dropped it and added an alias for the crawler.

yarikoptic commented:
> > @kyleam - did you spot any points which would somehow show that it wouldn't be possible?
> 
> Aside from the appveyor/direct mode failure being discussed elsewhere, I suppose? I think we should be fine, but we should know once we drop the root commit of the branch, right? That'll bring us back to the old annex.

yeap.

kyleam commented:
With d6dca0212 as the root, we're all green aside for the #2976 appveyor run and the known direct failures run (which has been red for a while now).  I'll drop the root commit.

kyleam commented:
> With [d6dca02](https://github.com/datalad/datalad/commit/d6dca021245607ea65481481e5a73aa65c461ddc) as the root, we're all green aside for the #2976 appveyor run and the known direct failures run (which has been red for a while now). I'll drop the root commit.

Travis is all green without the root commit.
